apps/backend/src/libs/modules/encryption/encryption.service.ts

```diff
@@ -0,0 +1,55 @@
+import bcrypt from "bcrypt";
+
+import { ApplicationError } from "~/libs/exceptions/exceptions.js";
+
+import { EncryptionServiceMessage } from "./libs/enums/enums.js";
+
+const DEFAULT_SALT_ROUNDS = 10;
+
+class EncryptionService {
+	private saltRounds: number;
+
+	public constructor() {
+		this.saltRounds = process.env["SALT_ROUNDS"]
+			? Number.parseInt(process.env["SALT_ROUNDS"])
+			: DEFAULT_SALT_ROUNDS;
+	}
+
+	public async comparePassword(
+		password: string,
+		hash: string,
+	): Promise<boolean> {
+		try {
+			return await bcrypt.compare(password, hash);
+		} catch (error) {
+			throw new ApplicationError({
+				cause: error,
+				message: EncryptionServiceMessage.COMPARE_PASSWORD,
+			});
+		}
+	}
+
+	public async generateSalt(): Promise<string> {
+		try {
+			return await bcrypt.genSalt(this.saltRounds);
+		} catch (error) {
+			throw new ApplicationError({
+				cause: error,
+				message: EncryptionServiceMessage.GENERATE_SALT,
+			});
+		}
```

Comment:
There is no need to wrap each async call into try catch, we will handle this on global or service level. The same for other methods here

```suggestion
	    return await bcrypt.genSalt(this.saltRounds);
```